src/Agent.Listener/Agent.cs

```diff
@@ -356,15 +357,28 @@ private async Task<int> RunAsync(AgentSettings settings, bool runOnce = false)
                     Task<bool> selfUpdateTask = null;
                     bool runOnceJobReceived = false;
                     jobDispatcher = HostContext.CreateService<IJobDispatcher>();
+
+                    // check if parallel polling should be prevented
+                    bool preventParallelPolling = AgentKnobs.PreventParallelPolling.GetValue(UtilKnobValueContext.Instance()).AsBoolean();
                     TaskAgentMessage previuosMessage = null;
 
+                    Task<TaskAgentMessage> getNextMessage = null;
```

Comment:
>getNextMessage [](http://example.com/codeflow?start=43&length=14)

this declaration can be inside the "try { block"

Reply:
I don't think so because you would reset this variable to null when it might have a value and we again would have the same problem. There's a reason I'm checking this value for null in the if statement later on.